pkg/cluster/validate.go

```diff
@@ -6,11 +6,16 @@ package cluster
 import (
 	"context"
 
+	"github.com/Azure/ARO-RP/pkg/util/azureclient/azuresdk/azcore"
 	"github.com/Azure/ARO-RP/pkg/validate"
 )
 
 func (m *manager) validateResources(ctx context.Context) error {
+	var clusterMSICredential azcore.TokenCredential
+	if m.doc.OpenShiftCluster.UsesWorkloadIdentity() {
+		clusterMSICredential = m.userAssignedIdentities.GetClusterMSICredential()
```

Comment:
I don't think this will work during an `az aro update` since the cluster MSI clients get initialized after dynamic validation. You'll want to move `m.ensureClusterMsiCertificate` and `m.initializeClusterMsiClients` to be before `m.validateResources`.

Reply:
I need a change in parallel to move MSI client instantiation before dynamic validation anyways, in order to validate existing federated identity credentials on new/existing identities. 

@rajdeepc2792 want me to push that commit onto your branch?

Reply:
https://github.com/Azure/ARO-RP/tree/tsatam/ARO-11854-handle-noncluster-federated-credentials it's literally the only change I have made on this branch so far.

Reply:
I have used the top commit from @tsatam  fork/branch as it is.